### DISCUSSION TITLE
Chardet Package Not Found

### DISCUSSION THREAD

**Discussion by wanmukhlis** [linuxdeepin/developer-center#5049] [Troubleshooting | 故障排查]
I tried to run 'make' command for deepin-pdfium but i got error 'chardet development package not found'. i already install chardet and test on python terminal but everything looks fine. please advice me
![1](https://github.com/linuxdeepin/developer-center/assets/29225381/8124ae82-feee-44e6-a9c3-44f27a0465ab)

**Comment by BLumia** [comment]
You need related development packages. Try `apt build-dep deepin-pdfium` to get required packages to build from source.

**Comment by wanmukhlis** [comment]
thanks for the reply. i tried to run apt build-dep deepin-pdfium but got this error.
![3](https://github.com/linuxdeepin/developer-center/assets/29225381/0a499325-5ccd-45bf-ada5-200cc6691be2)

**Reply by BLumia** [reply]
Check if you enabled the `deb-src` repo in your `/etc/apt/sources.list`. If not, duplicated existing rows which started with `deb` and modify the `deb` to `deb-src`, then do a `sudo apt update`, and then you can use `apt build-dep package-name` freely.

Alternatively you can also `sudo apt build-dep .` from the source folder (where you downloaded and uncompressed the source code).

**Comment by wanmukhlis** [comment]
thanks for the reply and it works! i managed to make the deepin-pdfium package. however when i tried to run sudo apt build-dep ./ on deepin dde-file-manager, i still got error of missing the pdfium-dev package. is it the different one?
![4](https://github.com/linuxdeepin/developer-center/assets/29225381/5d2c41d7-321c-4b2f-a7d8-efd24f8cece8)

**Reply by BLumia** [reply]
Did you build the `deepin-pdfium` *package* or only the binary files? If you are building `deepin-pdfium` so you can build `dde-file-manager`, then you need to build the deb *package*s.

And also, I'm not sure if you are on v20 or v23 beta, for the latter case, you should be able to get those packages without issues. If you are on v20, then you probably should checkout to the version that your system provides and then start building. The `master` branch are for v23 so there might be unmet dependencies if you are not on v23.
